File: lessons/compiled_lang/go/src/e40.branch.go
package main

import (
	"bufio"
	"fmt"
	"os"
)

func main() {
	fmt.Print("Select an item from the menu.\n\n" +
		"  1 - Coffee\n" +
		"  2 - Espresso\n" +
		"  3 - Latte\n" +
		"  4 - Machiato\n" +
		"  5 - Capucino\n" +
		"  6 - Mocha\n" +
		"  7 - Tea\n\n" +
		"Make your selection: ")

	reader := bufio.NewReader(os.Stdin)
	b, _ := reader.ReadByte()
	selection := int(b) - int('0')

	switch selection {
	case 1:
		fmt.Println("You selected a Coffee")
	case 2:
		fmt.Println("You selected an Espresso")
	case 3:
		fmt.Println("You selected a Latte")
	case 4:
		fmt.Println("You selected a Machiato")
	case 5:
		fmt.Println("You selected a Capucino")
	case 6:
		fmt.Println("You selected a Mocha")
	case 7:
		fmt.Println("You selected a Tea")
	default:
		fmt.Println("You have not entered a valid selection")
	}
}

File: lessons/gen_scripts/php/scripts/e40.branch.php
#!/usr/bin/env php
<?php
// PHP heredoc strips exactly the trailing newline right before the
//  closing marker, so this ends in "...Make your selection: " with no
//  linebreak before the answer is typed - no extra trim needed.
$menu = <<<EOT
Select an item from the menu.

  1 - Coffee
  2 - Espresso
  3 - Latte
  4 - Machiato
  5 - Capucino
  6 - Mocha
  7 - Tea

Make your selection: 
EOT;
echo $menu;
$selection = trim(fgets(STDIN));

switch ($selection) {
  case 1:
    echo "You selected a Coffee\n";
    break;
  case 2:
    echo "You selected an Espresso\n";
    break;
  case 3:
    echo "You selected a Latte\n";
    break;
  case 4:
    echo "You selected a Machiato\n";
    break;
  case 5:
    echo "You selected a Capucino\n";
    break;
  case 6:
    echo "You selected a Mocha\n";
    break;
  case 7:
    echo "You selected a Tea\n";
    break;
  default:
    echo "You have not entered a valid selection\n";
}
?>
